lib/internal/test_runner/test.js

```diff
@@ -834,15 +864,17 @@ class Suite extends Test {
 
       await this.runHook('before', hookArgs);
 
-      const stopPromise = stopTest(this.timeout, this.signal);
+      testTimeout = stopTest(this.timeout, this.signal);
       const subtests = this.skipped || this.error ? [] : this.subtests;
       const promise = SafePromiseAll(subtests, (subtests) => subtests.start());
 
-      await SafePromiseRace([promise, stopPromise]);
+      await SafePromiseRace([promise, testTimeout.promise]);
+      testTimeout.cleanup();
```

Comment:
move this to finally? (and convert to `testTimeout[SymbolDispose]()` after using `addAbortListener`)

Reply:
updated is this what you meant?

Reply:
if you meant to return a promise and on that promise add the `SymbolDispose` I think it's misleading as promises are not disposable

Reply:
Changed back to have a cleaner release to v18 and v16

Reply:
It wont get to v16 anyway, and I think `SymbolDispose` can be backported to 18.